desihub/desispec
# add QA+zmtl support

This PR adds support for combining tile_qa with zmtl creation (formerly zqso, owned by desitarget now ported to desispec).  This PR requires desihub/desitarget#754 (branch zwarn_badspecqa) to get two additional ZWARN bit definitions for BAD_SPECQA and BAD_PETALQA.

Example outputs are in /global/cfs/cdirs/desi/users/sjbailey/spectro/redux/zmtl , see README with details e.g.
```
[setup the prod, then...]
desi_tile_qa -n 20210606 -t 1230,1246,1394
make_zmtl_files -n 20210606 -t 1230
make_zmtl_files -n 20210606 -t 1246
make_zmtl_files -n 20210606 -t 1394
```

I added updates for running this as part of the pipeline but haven't tested that yet.

Let's also re-check the QA before merging.  e.g. https://data.desi.lbl.gov/desi/users/sjbailey/spectro/redux/zmtl/tiles/cumulative/1394/20210606/tile-qa-1394-thru20210606.png flags THRUFRAC_X in red, but unexpected went from 178 good fibers (below threshold) to 257 good fibers (above threshold) compared to @julienguy original run at https://data.desi.lbl.gov/desi/users/jguy/exposure_qa/tiles/cumulative/1394/20210606/tile-qa-1394-20210606.png that rejected the entire tile.

This PR does not:
  1. run desi_exposure_qa automatically at the time of cframes and propagate QAFIBERMASK forward into spectra/coadd/zbest FIBERMAP.FIBERSTATUS (it only gets run retroactively by desi_tile_qa at the end)
  2. set FIBERSTATUS bit POORPOSITION at the start along side BADPOSITION, independent of the QA.
  3. do anything with how tiles.csv is updated.

I'd like to do those soon, but they aren't strictly required for the initial round of zmtl updates for unlocking tiles.

@julienguy @akremin @geordie666
